Automattic/wp-calypso
# Adjustments to domain step in `onboarding-with-email` signup flow

#### Proposed Changes

Letero is working on a set of improvements to the `onboarding-with-email` signup flow. This flow is not yet actively used, but it will capture users that have shown interest in the email solutions offered on WordPress.com. For more context, see pbLl1t-1f6-p2.

In this PR, I have made a few changes to the domains step in that signup flow:
1. Added a variation of the domains step: `domains-with-email`
2. Hidden the free wp.com subdomain suggestion in that step. We do this because we can't offer users email at all if they don't pick a custom domain first.
3. Changed the heading in that step to `Choose a domain for your Professional Email`

#### Testing Instructions



**1️⃣ Test the onboarding-with-email flow**
1. Navigate to `/start/onboarding-with-email`
2. Search for a domain
3. Ensure that you are not presented with a `Free` wordpress.com domain suggestion
4. Ensure that the heading reads `Choose a domain for your Professional Email`
5. Ensure that there is no `Choose my domain later` button in the sidebar
6. Click `Use a domain I own` in the sidebar
7. Ensure that the page matches the screenshot below (except that the step counter should say "Step 1 of 3"):

![wordpress com_start_domains_use-your-domain(Medium laptop)](https://user-images.githubusercontent.com/1101677/186878714-0bd42755-1077-4146-901a-d21927be08ad.png)

**2️⃣ Test the regular flow for regressions**
1. Navigate to `/start`
2. Search for a domain
3. Ensure that you *are* presented with a `Free` wordpress.com domain suggestion
4. Ensure that the heading reads `Choose a domain`
5. Ensure that there *is* a `Choose my domain later` button in the sidebar

#### Pre-merge Checklist






Related to #
